xds/internal/xdsclient/xdsresource/errors.go

```diff
@@ -58,3 +58,14 @@ func ErrType(e error) ErrorType {
 	}
 	return ErrorTypeUnknown
 }
+
+// ErrResourceTypeUnsupported is an error used to indicate an unsupported xDS
+// resource type. The wrapped ErrStr contains the details.
+type ErrResourceTypeUnsupported struct {
+	ErrStr string
```

Comment:
This needs to be exported because this needs to be returned by the `authority` layer when the transport passes it resources of an unregistered type. The `authority` looks up the registry of resource types and if the URL is not found, it returns something like this:
```
return xdsresource.ErrResourceTypeUnsupported{ErrStr: fmt.Sprintf("Resource URL %v unknown in response from server", resourceUpdate.URL)}
```

Reply:
Got rid of the new error type and instead added a type to the existing enum. Now, the error string is not exported.